core/domain/story_domain.py

```diff
@@ -656,6 +656,25 @@ def require_valid_title(cls, title):
         if title == '':
             raise utils.ValidationError('Title field should not be empty')
 
+    def get_acquired_skill_ids_for_exp_ids(self, exp_ids):
+        """Returns the acquired skill ids of the nodes having the given
+        exploration ids.
+
+        Args:
+            exp_ids: list(str). The list of IDs of the explorations linked to
+            the story.
+
+        Returns:
+            list(str)|None. The list of acquired skill ids for the
+                explorations or None, if no node is linked to them.
+        """
+        acquired_skill_ids = []
+        for node in self.story_contents.nodes:
```

Comment:
I think we can should be able to get the acquired skill ids from the exploration itself no? (@aks681 / @seanlip do you know?) Right now you are doing a linear search between the passed exploration ids and all the exploration ids in the story node, I am worried this may get expensive

Reply:
This is a question for @aks681 I think. (I don't think skills are in the exploration but they should be in the story nodes.)

Reply:
Yes I agree it would make more sense to get them by story nodes. Changed that just now.

Reply:
Yeah, story nodes have skill ids in them, but not explorations.